### PR TITLE
feat(utils): now `origin` only requires the `feature`

### PR DESCRIPTION
**Pull request by tiborux**
This PR removes the `location` to be not required. Only the feature will be required.

EX-5077

### CODE REVIEW

File: packages/x-components/src/utils/__tests__/origin.spec.ts [hunk 1]
```diff
@@ -22,12 +16,21 @@ describe(`testing ${createOrigin.name} utility method`, () => {
     ).toBeNull();
   });
 
-  it('returns `feature:location` when the feature and the location are provided', () => {
+  it('returns `feature:location` when the feature is provided', () => {
```
Comment (pmareke) [inline packages/x-components/src/utils/__tests__/origin.spec.ts:19]: I don't think this title is ok, as you are saying `the feature is provided` but the  in this case the location is relevant as it's part of the final origin.

File: packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts [hunk 2]
```diff
@@ -159,9 +159,6 @@ describe('testing identifier results module actions', () => {
       await store.dispatch('saveOrigin', { location: 'predictive_layer' });
       expect(store.state.origin).toBeNull();
 
-      await store.dispatch('saveOrigin', { feature: 'search_box' });
```
Comment (pmareke) [inline packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts:162]: Why did you remove it? Please check the result of this case.

File: packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts [hunk 3]
```diff
@@ -337,9 +337,6 @@ describe('testing search module actions', () => {
       await store.dispatch('saveOrigin', { location: 'predictive_layer' });
       expect(store.state.origin).toBeNull();
 
-      await store.dispatch('saveOrigin', { feature: 'search_box' });
```
Comment (pmareke) [inline packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts:340]: Same here.

**Review by pmareke — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts (hunk 2 above)
Comment (tiborux) [reply packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts:162]: I removed it because right now It makes no sense because you can create an origin with only the feature and this test is to check when it's null.

File: packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts (hunk 3 above)
Comment (tiborux) [reply packages/x-components/src/x-modules/search/store/__tests__/actions.spec.ts:340]: I removed it because right now It makes no sense because you can create an origin with only the feature and this test is to check when it's null.

**Review by CachedaCodes — APPROVED** [review]

File: packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts (hunk 2 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/identifier-results/store/__tests__/actions.spec.ts:162]: But maybe you can create another test but not remove this validation right?

Now you have three different scenarios when you dispatch the action:

- Location
- Feature
- Location + Feature

But you're only testing two of them right?

**Review by pmareke — APPROVED** [review]

**Review by CachedaCodes — APPROVED** [review]
